src/libraries/System.Private.CoreLib/src/System/Collections/HashHelpers.cs

```diff
@@ -57,11 +57,9 @@ public static int GetPrime(int min)
             if (min < 0)
                 throw new ArgumentException(SR.Arg_HTCapacityOverflow);
 
-            foreach (int prime in s_primes)
-            {
-                if (prime >= min)
-                    return prime;
-            }
+
+            if (TryBinarySearchForPrime(min, out int prime))
+                return prime;
```

Comment:
This is going to penalize _the_ most important case, small values, and in particular 0, for when a dictionary is created without an initial capacity.